doc/guides/device_mgmt/smp_protocol.rst

```diff
@@ -0,0 +1,289 @@
+    | ``3``         | Application/system configuration              |
+    |               | (currently not used by Zephyr)                |
+    +---------------+-----------------------------------------------+
+    | ``4``         | Application/system log management;            |
+    |               | (currently not used by Zephyr)                |
+    +---------------+-----------------------------------------------+
+    | ``5``         | Designated for test crashing application      |
+    |               | (unused by Zephyr)                            |
+    +---------------+-----------------------------------------------+
+    | ``6``         | Designated for run tests                      |
+    |               | (unused by Zephyr)                            |
+    +---------------+-----------------------------------------------+
+    | ``7``         | Split image management                        |
+    |               | (unused by Zephyr)                            |
+    +---------------+-----------------------------------------------+
+    | ``8``         | :ref:`mcumgr_smp_group_8`                     |
+    +---------------+-----------------------------------------------+
+    | ``9``         | :ref:`mcumgr_smp_group_9`                     |
+    +---------------+-----------------------------------------------+
+    | ``63``        | Zephyr specific basic commands group          |
+    +---------------+-----------------------------------------------+
+    | ``64``        | This is the base group for defining           |
+    |               | an application specific management groups     |
+    +---------------+-----------------------------------------------+
+
+The payload for above groups, except for ``64`` which is not defined,
+is always CBOR encoded.
+
+Data format and minimal response
+********************************
+
+SMP data is always sent in the CBOR map format; this can be either finite
+or infinite map, the SMP protocol does not specify which kind should be
+used where.
+
+Regardless of a command issued, as long as there is SMP client on the
+other side of request, a response should at least contain operation
+status/error code; lack of response is only allowed when there is
+no SMP service or device is non-responsive.
+
+Minimal response
+================
+
+Minimal response is CBOR directory
+
+.. code-block:: none
+
+    {
+        (string)"rc" : (int)<rc_code>
+    }
+
+where ``<rc_code>`` takes value from :ref:`mcumgr_smp_protocol_status_codes`
+
+.. _mcumgr_smp_protocol_status_codes:
+
+Status/error codes in responses
+===============================
+
+.. table::
+    :align: center
+
+    +---------------+-----------------------------------------------+
+    | Decimal ID    | Meaning                                       |
+    +===============+===============================================+
+    | ``0``         | No error, OK                                  |
```

Comment:
Worth putting the define names here also e.g. MGMT_ERR_EOK?

Reply:
I would rather leave that as general SMP documentation with Zephyr specific defines left out. People looking into implementing SMP client may not be really interested how Zephyr calls them, just need a way to define frames in their language of choice.
Anyway, the headers inside lib need to be altered to include doxygen tagging so that we could generate API documentation for it.